lms/djangoapps/student_account/views.py

```diff
@@ -343,32 +342,28 @@ def get_user_orders(user):
     )
 
     for order in commerce_user_orders:
-        try:
-            if order['status'].lower() == 'complete':
-                order_lines = order['lines']
-                date_placed = datetime.strptime(order['date_placed'], "%Y-%m-%dT%H:%M:%SZ")
-                order_data = {
-                    'number': order['number'],
-                    'price': order['total_excl_tax'],
-                    'order_date': strftime_localized(
-                        date_placed.replace(tzinfo=pytz.UTC), 'SHORT_DATE'
-                    ),
-                    'receipt_url': EcommerceService().get_receipt_page_url(order['number']),
-                    'lines': order_lines
-                }
-
-                if len(order_lines) == 1:
-                    product = order_lines[0].get('product')
-
-                    if product:
-                        for attribute in product['attribute_values']:
-                            if attribute['name'] == 'certificate_type' and attribute['value'] in allowed_course_modes:
-                                user_orders.append(order_data)
-                else:
-                    user_orders.append(order_data)
-        except KeyError:
-            log.exception('Invalid order structure: %r', order)
-            return no_data
+        if order['status'].lower() == 'complete':
+            date_placed = datetime.strptime(order['date_placed'], "%Y-%m-%dT%H:%M:%SZ")
+            order_data = {
+                'number': order['number'],
+                'price': order['total_excl_tax'],
+                'order_date': strftime_localized(
+                    date_placed.replace(tzinfo=pytz.UTC), 'SHORT_DATE'
+                ),
+                'receipt_url': EcommerceService().get_receipt_page_url(order['number']),
+                'lines': [],
+            }
+
+            # If the order lines contain a product that is not a Seat
```

Comment:
@iivic I agree.  Coupons are only allowed to be created by admins at edX and sometime in the future self service.  So having them show up on the order history is probably a good thing.  Just remove the filter and show all line items.